### PR TITLE
CLI-684: Automate CCB ticket creation for pre-releases.

### PR DESCRIPTION
**Pull request by grasmash**
**Proposed changes**

Automatically creates a CCB ticket whenever a pre release is created.

**Alternatives considered**

**Testing steps**

1. Create a pre-release.
2. Check JIRA.

OR, test script locally:
```
export GITHUB_RELEASE_BODY="test body"; export GITHUB_RELEASE_NAME="1.0.0"; export JIRA_HOST="[url]"; export GITHUB_ACTIONS_RUN_URL="[url]"; export PERSONAL_ACCESS_TOKEN="[token]"; export TOKEN_BASED_AUTH="true"; ./.github/workflows/create-ccb-ticket.sh
```

**Merge requirements**
- [x] _Bug_, _enhancement_, or _breaking change_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

**Comment by danepowell** [comment]
I added the Jira user secrets to the repo

**Comment by grasmash** [comment]
@danepowell I've validated that create-ccb-ticket.sh works but I have not tested release.yml. I think I need to actually create a release for that?

File: .github/workflows/create-issue.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+on: push
+  issues:
+    types: [opened]
```
Comment (grasmash) [inline .github/workflows/create-issue.yml:3]: @danepowell This is unrelated to release. Just replaces current Zapier functionality.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          JIRA_USER_EMAIL: ${{ secrets.JIRA_USER_EMAIL }}
+      - name: Create CCB ticket
+        uses: acquia/gajira-create@server
+        with:
+          project: DX
+          issuetype: Release
+          summary: ${{ github.events.issue.title }}
+          description: ${{ env.JIRA_DESCRIPTION }}
+          # @todo Replace with CLI component ID.
+          fields: '{"components": [{"id": "21989"}]}'
```
Comment (grasmash) [inline .github/workflows/release.yml:32]: @danepowell Do you know the right ID for ACLI?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    runs-on: ubuntu-latest
+    name: Create a CCB ticket in Jira
+    steps:
+      - name: Generate body
+        env:
+          # @see https://docs.github.com/en/rest/reference/releases#get-a-release
+          GITHUB_RELEASE_BODY: ${{ github.event.release.body }}
+          GITHUB_RELEASE_NAME: ${{ github.event.release.name }}
+          GITHUB_ACTIONS_RUN_URL: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
+        run: export JIRA_DESCRIPTION="$(${GITHUB_WORKSPACE}/.github/workflows/create-ccb-ticket.sh)"
```
Comment (grasmash) [inline .github/workflows/release.yml:16]: Hopefully this export actually works and passed to the next step.

File: .github/workflows/release.yml (hunk 2 above)
Comment (anavarre) [reply .github/workflows/release.yml:34]: I'm not Dane but I can help you with that. It's at `<JIRA_BASE_URL>/rest/api/latest/project/CLI`

File: .github/workflows/create-issue.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+          JIRA_USER_EMAIL: ${{ secrets.JIRA_USER_EMAIL }}
+      - name: Create Jira ticket
+        id: create
+        uses: acquia/gajira-create@server
+        with:
+          project: DX
+          issuetype: Task
+          summary: ${{ github.events.issue.title }}
+          description: ${{ github.events.issue.url }}
+          fields: '{"components": [{"id": "21989"}]}'
```
Comment (grasmash) [inline .github/workflows/create-issue.yml:24]:
```suggestion
          fields: '{"components": [{"id": "20084"}]}'
```

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      - name: Create CCB ticket in Jira
+        id: create
+        uses: acquia/gajira-create@server
+        with:
+          project: DX
+          issuetype: Release
+          summary: ${{ github.events.issue.title }}
+          description: ${{ env.JIRA_DESCRIPTION }}
+          # @todo Replace with CLI component ID.
+          fields: '{"components": [{"id": "21989"}]}'
```
Comment (grasmash) [inline .github/workflows/release.yml:34]:
```suggestion
          fields: '{"components": [{"id": "20084"}]}'
```

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+          JIRA_USER_EMAIL: ${{ secrets.JIRA_USER_EMAIL }}
+      - name: Create CCB ticket in Jira
+        id: create
+        uses: acquia/gajira-create@server
+        with:
+          project: DX
+          issuetype: Release
+          summary: ${{ github.events.issue.title }}
+          description: ${{ env.JIRA_DESCRIPTION }}
+          # @todo Replace with CLI component ID.
```
Comment (grasmash) [inline .github/workflows/release.yml:33]:
```suggestion
```

File: .github/workflows/create-issue.yml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+      - name: Log created issue
+        run: echo "Issue ${{ steps.create.outputs.issue }} was created"
+      - name: Update Github issue with Jira ticket prefix
+        uses: actions-cool/issues-helper@v3
```
[the remark this answers is not in the document]
Comment (danepowell) [reply .github/workflows/create-issue.yml:28]: So if I create a new issue "Add cool feature foo", this will create a Jira ticket, then automatically change the title to "CLI-123: Add cool feature foo"? Pretty slick! But I feel like this shouldn't require a third-party action. Have you looked into whether Actions can manage issues natively?

I just worry about getting approval for that particular action.

[the remark this answers is not in the document]
Comment (grasmash) [reply .github/workflows/create-issue.yml:28]: Yes, that's what it does. 

I couldn't find a native action to do this.

[the remark this answers is not in the document]
Comment (danepowell) [reply .github/workflows/create-issue.yml]: I suggested a change to avoid using a third-party action. See also https://docs.github.com/en/actions/using-workflows/using-github-cli-in-workflows

You'll probably need to set issue write permissions on the token, same as perform.

File: .github/workflows/create-issue.yml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  issues:
+    types: [opened]
+name: Create new tickets in Jira
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Create new tickets in Jira
+    steps:
+      - name: Login to Jira
+        uses: acquia/gajira-login-2@server
```
Comment (danepowell) [inline .github/workflows/create-issue.yml:11]:
```suggestion
        uses: acquia/gajira-login@server
```

File: .github/workflows/create-issue.yml [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+          JIRA_API_TOKEN: ${{ secrets.JIRA_API_TOKEN }}
+          JIRA_USER_EMAIL: ${{ secrets.JIRA_USER_EMAIL }}
+      - name: Create Jira ticket
+        id: create
+        uses: acquia/gajira-create@server
+        with:
+          project: DX
+          issuetype: Task
+          summary: ${{ github.events.issue.title }}
+          description: ${{ github.events.issue.url }}
```
Comment (danepowell) [inline .github/workflows/create-issue.yml:23]:
```suggestion
          summary: ${{ github.event.issue.title }}
          description: ${{ github.event.issue.url }}
```

**Comment by danepowell** [comment]
@grasmash let's split the creation of CCB and non-CCB tickets. I've got non-CCB tickets working in #853

File: .github/workflows/release.yml [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+          JIRA_BASE_URL: ${{ secrets.JIRA_BASE_URL }}
+          JIRA_API_TOKEN: ${{ secrets.JIRA_API_TOKEN }}
+          JIRA_USER_EMAIL: ${{ secrets.JIRA_USER_EMAIL }}
+      - name: Create CCB ticket in Jira
+        id: create
+        uses: acquia/gajira-create@server
+        with:
+          project: CLI
+          issuetype: Release
+          summary: ${{ github.events.issue.title }}
```
Comment (danepowell) [inline .github/workflows/release.yml:32]:
```suggestion
          summary: ${{ github.event.issue.title }}
```

File: .github/workflows/release.yml [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+          description: ${{ env.JIRA_DESCRIPTION }}
+          fields: '{"components": [{"id": "20084"}]}'
```
Comment (danepowell) [inline .github/workflows/release.yml:34]:
```suggestion
          fields: '{"components": [{"id": "25474"}]}'
```

**Review by danepowell — APPROVED** [review]
